hypervisor/arch/x86/assign.c

```diff
@@ -791,7 +791,7 @@ int ptdev_msix_remap(struct vm *vm, uint16_t virt_bdf,
 	/* build physical config MSI, update to info->pmsi_xxx */
 	ptdev_build_physical_msi(vm, info, dev_to_vector(entry->node));
 	entry->ptdev_intr_info.msi = *info;
-	entry->ptdev_intr_info.msi.virt_vector = info->vmsi_data & 0xFF;
+	entry->ptdev_intr_info.msi.virt_vector = (int)(info->vmsi_data & 0xFF);
```

Comment:
@junjiemao1  is revisiting the entire code on the usage of vector and plan to use unsigned int to replace them. uint16_t... This is WIP, so I will pause this patch for now,,,